tools/IoTEdgeDevOps/DevOpsLibTest/BuildManagementTest.cs

```diff
@@ -32,69 +34,159 @@ public void TearDown()
         }
 
         [Test]
-        public async Task TestGetLatestBuildAsync()
+        public async Task TestGetLatestBuildsAsyncWithNoDefinitionId()
+        {
+            ArgumentException ex = Assert.ThrowsAsync<ArgumentException>(
+                async () => { await this.buildManagement.GetLatestBuildsAsync(new int[] { }, "anybranch").ConfigureAwait(false); });
+
+            Assert.True(ex.Message.StartsWith("Cannot be null or empty collection."));
+            Assert.That(ex.ParamName, Is.EqualTo("buildDefinitionIds"));
+        }
+
+        [Test]
+        public async Task TestGetLatestBuildsAsyncWithEmptyBranchName()
+        {
+            ArgumentException ex = Assert.ThrowsAsync<ArgumentException>(
+                async () => { await this.buildManagement.GetLatestBuildsAsync(BuildDefinitionIds.MasterBranchReporting, " ").ConfigureAwait(false); });
+
+            Assert.True(ex.Message.StartsWith("Cannot be null or white space."));
+            Assert.That(ex.ParamName, Is.EqualTo("branchName"));
+        }
+
+        [Test]
+        public async Task TestGetLatestBuildsAsyncWithBuildResults()
         {
             this.httpTest.RespondWithJson(
                 new
                 {
-                    buildNumber = "20191019.2",
-                    _links = new
+                    count = 2,
+                    value = new object[]
                     {
-                        web = new {
-                            href = "https://dev.azure.com/msazure/b32aa71e-8ed2-41b2-9d77-5bc261222004/_build/results?buildId=25980152"
+                        new
+                        {
+                            definition = new
+                            {
+                                id = BuildDefinitionIds.BuildImages
+                            },
+                            buildNumber = "20191031.6",
+                            _links = new
+                            {
+                                sourceVersionDisplayUri = new
+                                {
+                                    href = "https://dev.azure.com/msazure/b32aa71e-8ed2-41b2-9d77-5bc261222004/_apis/build/builds/26326316/sources"
+                                },
+                                web = new
+                                {
+                                    href = "https://dev.azure.com/msazure/b32aa71e-8ed2-41b2-9d77-5bc261222004/_build/results?buildId=26326316"
```

Comment:
1. Isn't `b32aa71e-8ed2-41b2-9d77-5bc261222004` our subscription ID? I'm not sure if that should be exposed? 
2. Did you make sure the `buildId=26326316` would always be there for testing?

Reply:
buildId=26326316 is just a mock data; you can mock whatever you want. I think subscription id is not secret, I can change it; but no one can use subscription to access our azure resources.  right?